src/app/Dashboard/LayoutTemplateUploadModal.tsx

```diff
@@ -71,53 +79,66 @@ export const DashboardLayoutUploadModal: React.FC<DashboardLayoutUploadModalProp
   const [allOks, setAllOks] = React.useState(false);
   const [uploading, setUploading] = React.useState(false);
 
-  React.useEffect(() => {
-    if (allOks && numOfFiles) {
-      const latest = dashboardConfigs.layouts[dashboardConfigs.layouts.length - 1];
-      dispatch(dashboardConfigReplaceLayoutIntent(latest.name));
-    }
-  }, [dispatch, dashboardConfigs, allOks, numOfFiles]);
-
-  const parseDashboardLayoutFile = React.useCallback(
-    (file: File): Observable<DashboardLayout> => {
+  const validateParseTemplate = React.useCallback(
+    (file: File): Observable<LayoutTemplate> => {
       return from(
         file.text().then((content) => {
-          let layout: DashboardLayout;
+          let template: SerialLayoutTemplate;
```

Comment:
I did these:

1. Download a layout as template.
2. Upload the template.
3. Failed to upload with error `Invalid`.

![Screenshot from 2023-04-12 21-12-25](https://user-images.githubusercontent.com/68053619/231620149-79f881c4-44fa-484e-8553-6039fcd8e521.png)


Template content:

```json
{"name":"Custom123","description":"asdasdasd","cards":[],"version":"v2.3"}
```

Reply:
Oh empty template is not allowed?

Reply:
```ts
if (!Array.isArray(template.cards) || template.cards.length === 0) {
  throw new Error(t('LayoutTemplateUploadModal.ERROR.CONFIG_INVALID'));
}
```